components/ai_chat/core/common/mojom/ai_chat.mojom

```diff
@@ -232,10 +232,10 @@ struct ToolUseEvent {
   // ID of the tool use request
   string id;
 
-  // Arguments for the Tool. The LLM may hallucinate and generate
-  // incorrect arguments or even invalid JSON, so this should be
-  // interpreted defensively.
-  mojo_base.mojom.DictionaryValue arguments;
+  string arguments_json;
```

Comment:
let's note that this is the source of truth and that the below is just for convenience/caching